serverless/ingest.py

```diff
@@ -0,0 +1,197 @@
+    count = 0
+    for question in questions:
+        cursor.execute("SELECT id FROM \"Question\" WHERE title = %s", [question.title])
+        result = cursor.fetchone()
+        if (result is not None):
+            logging.info("Existing question: %s has ID %s." % (question.title, result[0]))
+            continue
+        category_id = categories[question.category]
+        cursor.execute("INSERT INTO \"Question\" (title, difficulty, \"categoryId\", description, hints) VALUES (%s, %s, %s, %s, %s) RETURNING id",
+                       [question.title, question.difficulty, category_id, question.description, question.hints])
+        question_id = cursor.fetchone()[0]
+        for topic in question.topics:
+            topic_id = topics[topic]
+            cursor.execute("INSERT INTO \"_QuestionToTopic\" (\"A\", \"B\") VALUES (%s, %s)",
+                           [question_id, topic_id])
+        cursor.execute("INSERT INTO \"TestCases\" (inputs, output, \"questionId\") VALUES (%s, %s, %s)",
+                       [question.testcase.inputs, question.testcase.output, question_id])
+        for language in question.templates:
+            cursor.execute("INSERT INTO \"SolutionTemplate\" (language, code, \"questionId\") VALUES (%s, %s, %s)",
+                           [language.name.upper(), question.templates[language], question_id])
+        count += 1
+        logging.info("Inserted question: %s has ID %s." % (question.title, question_id))
+    return count
+
+
+def process(data):
+    questions = []
+    for entry in data:
+        questions.append(Question(entry))
+    try:
+        host = environment["DATABASE_HOST"]
+        port = environment["DATABASE_PORT"]
+        database = environment["DATABASE_NAME"]
+        user = environment["DATABASE_USERNAME"]
+        password = environment["DATABASE_PASSWORD"]
+    except Exception as error:
+        logging.error("Database connection settings unavailable. (%s)" % error)
+        exit(1)
+    try:
+        with connect(host=host, port=port, dbname=database, user=user, password=password) as connection:
+            with connection.cursor() as cursor:
+                [categories, new_category_count] = (process_categories(questions, cursor))
+                [topics, new_topic_count] = (process_topics(questions, cursor))
+                new_question_count = process_questions(questions, cursor, categories, topics)
+                logging.info("[Summary] New categories: %s | New topics: %s | New questions: %s" %
+                             (new_category_count, new_topic_count, new_question_count))
+    except Exception as error:
+        logging.info(error)
+
+
+def usage(name: str):
+    logging.warn("Usage: %s <file> [<environment file>]\n" % (name))
+
+
+if __name__ == "__main__":
+    logging.basicConfig(
+        format="[%(process)d] %(asctime)s (%(levelname)s): %(message)s",
+        level=logging.INFO
+    )
+    if len(sys.argv) < 2 or len(sys.argv) > 3:
```

Comment:
Can we have some instructions on the top of the script on how to operate lines 183 to 190? There seems to be some deferring sys.argv

Reply:
I've added a README into the `serverless` folder.